src/main/kotlin/io/curity/dynamoDBDataAccessProvider/DynamoDBClient.kt

```
@@ -51,51 +51,99 @@ import software.amazon.awssdk.services.sts.StsClient
+                logger.warn("EC2 Instance ")
+                creds = Optional.of(InstanceProfileCredentialsProvider.builder().build())
+            } else if (accessMethod.accessKeyIdAndSecret.isPresent) {
+                logger.warn("Access Key")
+                val keyIdAndSecret = accessMethod.accessKeyIdAndSecret.get()
+                creds = Optional.of(StaticCredentialsProvider.create(AwsBasicCredentials.create(keyIdAndSecret.accessKeyId.get(), keyIdAndSecret.accessKeySecret.get())))
+
+                /* roleARN is present, get temporary credentials through AssumeRole */
+                if (keyIdAndSecret.awsRoleARN.isPresent) {
+                    creds = Optional.of(getNewCredentialsFromAssumeRole(creds.get(), keyIdAndSecret.awsRoleARN.get()))
+                }
+            } else if (accessMethod.aWSProfile.isPresent && accessMethod.aWSProfile.get().awsProfileName.isPresent) {
+                logger.warn("AWS Profile")
+                val awsProfile = accessMethod.aWSProfile.get()
+                creds = Optional.of(ProfileCredentialsProvider.builder()
+                        .profileName(awsProfile.awsProfileName.get())
+                        .build())
+
+                /* roleARN is present, get temporary credentials through AssumeRole */
+                if (awsProfile.awsRoleARN.isPresent) {
+                    creds = Optional.of(getNewCredentialsFromAssumeRole(creds.get(), awsProfile.awsRoleARN.get()))
+                }
+            } else if (accessMethod.aWSDirect.isPresent) {
+                logger.info("Configuring Direct connection")
+                val aWSDirect = accessMethod.aWSDirect.get()
+                creds = Optional.of(StaticCredentialsProvider.create(AwsBasicCredentials.create(aWSDirect.accessKeyId, aWSDirect.accessKeySecret)))
+            } else {
+                logger.info("empty - configuration issue")
+                creds = Optional.empty<AwsCredentialsProvider>()
```

I think it would be better to throw an exception here. The plugin is useless without a properly configured dynamo db client anyway. If you threw an exception here, then the `creds` doesn't have to be an optional and this method can return `DynamoDbClient` instead of an optional.